src/mbed_tools/devices/device.py

```diff
@@ -28,6 +29,35 @@ class Device:
     serial_number: str
     serial_port: Optional[str]
     mount_points: Tuple[Path, ...]
+    mbed_enabled: bool = False
+
+    @classmethod
+    def from_candidate(cls, candidate: CandidateDevice) -> "Device":
+        """Contruct a Device from a CandidateDevice.
+
+        We try to resolve a board using data files that may be stored on the CandidateDevice.
+        If this fails we set the board to `None` which means we couldn't verify this Device
+        as being an Mbed enabled device.
+
+        Args:
+            candidate: The CandidateDevice we're using to create the Device.
+        """
+        device_file_info = read_device_files(candidate.mount_points)
```

Comment:
In commit "devices: Add a device factory classmethod", flake8 erred:
```
src/mbed_tools/devices/device.py:45:9: F841 local variable 'device_file_info' is assigned to but never used
src/mbed_tools/devices/device.py:45:28: F821 undefined name 'read_device_files'
```

pytest also complains:
```
E       NameError: name 'read_device_files' is not defined
```